### PR TITLE
DSD-361: List listItems data prop

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes [DSD-361](https://jira.nypl.org/browse/DSD-361)

## **This PR does the following:**
- This adds the ability to render a `List` component through an array of data instead of through children elements.
- The `listItems` prop is used to render the correct children elements.
- Instead of passing `li` elements for `ListType.Unordered` or ListType.Ordered` types, `listItems` takes an array of strings such as `["fish", "cat", "dog"]`.
- Instead of passing `dt` and `dd` elements for `ListType.Definition` type, `listItems` takes an array of objects, each containing `term` and `definition` keys  such as `[{ term: "fish", definition: "swims" }, { term: "cat", definition: "meows" }, { term: "dog", definition: "barks" }]`.

### Front End Review:
- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/List/List.stories.mdx [hunk 1]
```diff
@@ -166,3 +171,85 @@ import { listRenderer, cardsList } from "./List.stories.tsx";
...
+  <dd>The mahi-mahi is an ocean fish known...</dd>
+  <dt>Golden trout</dt>
+  <dd>The golden trout is a beautifully colored fish...</dd>
+  <dt>Rainbowfish</dt>
+  <dd>The rainbowfish is a tropical fish known...</dd>
+  <dt>Suckerfish</dt>
+  <dd>The suckerfish is a curious fish that...</dd>
+</List>
+
+// With `itemList` data prop
```
Comment (bigfishdesign13) [inline src/components/List/List.stories.mdx:218]: Example code needs to be cleaned up (missing commas and brackets).

File: src/components/List/List.stories.mdx [hunk 2]
```diff
@@ -166,3 +171,85 @@ import { listRenderer, cardsList } from "./List.stories.tsx";
     )}
   </Story>
 </Preview>
+
+## Lists with data props
```
Comment (bigfishdesign13) [inline src/components/List/List.stories.mdx]: Nit: match the captialization used in other stories

File: src/components/List/List.stories.mdx [hunk 3]
```diff
@@ -166,3 +171,85 @@ import { listRenderer, cardsList } from "./List.stories.tsx";
...
+<List
+  type={ListTypes.Definition}
+  title="Animal Crossing Fish"
+  listItems={fishDefinitions}
+/>
+```
+
+<Preview>
+  <Story
+    name="List with data props"
```
Comment (bigfishdesign13) [inline src/components/List/List.stories.mdx]: Nit: match the captialization used in other stories

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
This looks great except for a few minor formatting issues.

**Comment by EdwinGuzman** [comment]
Good catches. Updated.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good.

**Comment by bigfishdesign13** [comment]
@EdwinGuzman I just realized there was no entry in the CHANGELOG for this PR.  Let's make sure to get something in there befor the release on Thursday.

**Pull request merged**